src/Kestrel.Core/Internal/Http/Frame.FeatureCollection.cs

```diff
@@ -321,5 +322,26 @@ void IHttpRequestLifetimeFeature.Abort()
         {
             Abort(error: null);
         }
+
+        async Task IHttpSendFileFeature.SendFileAsync(string path, long offset, long? count, CancellationToken cancellation)
+        {
+            ResponseHeaders.ContentLength = count;
```

Comment:
So the idea is that SendFileAsync takes over the whole response?

Reply:
Leaving this up to the caller, makes it usable in more cases.

The caller needs to figure out the length of the file, so perhaps a `FileStream` instead of string path argument makes more sense?

Reply:
"Range Requests": https://tools.ietf.org/html/rfc7233
Probably not a major use-case for Kestrel, but would be nice if it could be implemented using `IHttpSendFileFeature`.

Reply:
I'm not sure this belongs at the send file feature level. It's currently implemented in higher level components.

Reply:
I meant `IHttpSendFileFeature` should not set ContentLength and leave it up to the higher level components. e.g. to implement multi-range responses as described in the rfc.

Reply:
Ah, yes I agree

Reply:
> "Range Requests"

StaticFiles will use `IHttpSendFileFeature` for [these if available](https://github.com/aspnet/StaticFiles/blob/1e0fe87c204b0199b97bc1d74f095e9746e9e674/src/Microsoft.AspNetCore.StaticFiles/StaticFileContext.cs#L369-L374)

Reply:
I removed the header; but its at a low level; so it needs to know if chunking and then chunk ;-/

Reply:
nvm `StaticFiles` sets the header before calling `SendFile` so this works